repositories/memory/topics/list_topics.go

```diff
@@ -0,0 +1,17 @@
+package topics
+
+import (
+	"context"
+
+	"github.com/matheusmosca/walrus/domain/entities"
+)
+
+func (r *repository) ListTopics(_ context.Context) ([]entities.Topic, error) {
+	topics := make([]entities.Topic, 0, len(r.storage))
```

Comment:
If the storage length is 0, then an error should be returned. You can create an error in the domain layer like ErrNoTopicsFound and return it here. You can also use this error on unit tests asserting it with ErrorIs

Reply:
do you think an empty list should return an error?

Reply:
I thought about error reasons like (some problem with SGBDs), but if you think it's better I can return the error to empty list